lint/clang_tidy_wrapper.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/bash
+# This is a wrapper for clang-tidy which allows it to be run on a collection of files
+# Usage: clang-tidy-target.bash <clang-tidy-path> <file1> <file2> ... -- <compiler-args>
```

Comment:
nit: .bash -> .sh? (actually I think `.bash` is the correct extension for this file if it doesn't work under other shells)

Reply:
Fixed